index.js

```diff
@@ -319,35 +319,19 @@ module.exports = {
     }
 
     function listMembers(groupId, opts = {}) {
-      const deferedSource = pullDefer.source()
-
-      get(groupId, (err, group) => {
-        // prettier-ignore
-        if (err) return deferedSource.abort(clarify(err, 'Failed to get group info when listing members'))
-        // prettier-ignore
-        if (group.excluded) return deferedSource.abort( new Error("We're excluded from this group, can't list members"))
-
-        const source = pull(
-          ssb.db.query(
-            where(
-              and(
-                isDecrypted('box2'),
-                type('group/add-member'),
-                groupRecp(groupId)
-              )
-            ),
-            opts.live ? live({ old: true }) : null,
-            toPullStream()
-          ),
-          pull.map((msg) => msg.value.content.recps.slice(1)),
-          pull.flatten(),
-          pull.unique()
-        )
-
-        deferedSource.resolve(source)
-      })
-
-      return deferedSource
+      return pull(
+        getPickedEpoch(groupId, { live: !!opts?.live }),
+        pull.map((pickedEpoch) =>
+          getMemberUpdates(pickedEpoch.id, { live: !!opts?.live })
+        ),
+        pull.flatten(),
```

Comment:
I honestly don't know what flaten does if it's operating on streams... especially if the streams are "live"... if there are e.g. 2 epochs live, does the second epoch ever get a word in ?

Reply:
noo it fails. nice catch

Reply:
fixed this with a cool pull.take. things are a bit flaky though, see other thread